Sources/Security.php

```diff
@@ -238,6 +239,94 @@ public static function spamProtection(string $error_type, bool $only_return_resu
 		return false;
 	}
 
+	/**
+	 * Checks for the existence and security status of specific files and directories
+	 * required for the proper functioning of the system. Ensures that security measures
+	 * are applied and generates a list of warnings for any issues detected.
+	 *
+	 * Warnings include:
+	 * - Missing or insecure critical files (e.g., install.php, upgrade.php).
+	 * - Directories not properly secured (e.g., cache directory).
+	 * - Missing legal agreement or policy documents.
+	 * - Missing authentication secrets.
+	 *
+	 * @return bool|array Returns an array of warnings if any security risks are detected,
+	 *    or false if the user lacks the necessary permissions or is a guest.
+	 */
+	public static function checkSecurityFiles(): bool|array
+	{
+		if (User::$me->allowedTo('admin_forum') && !User::$me->is_guest) {
+			$security_files = [
+				'install.php',
+				'upgrade.php',
+				'convert.php',
+				'repair_paths.php',
+				'repair_settings.php',
+				'Settings.php~',
+				'Settings_bak.php~',
+			];
+
+			// Add your own files.
+			IntegrationHook::call('integrate_security_files', [&$security_files]);
+
+			// Filter out missing security files
+			$security_files = array_filter($security_files, function ($file) {
+				return file_exists(Config::$boarddir . '/' . $file);
+			});
+
+			// Determine attachment upload directory path
+			$path = !empty(Config::$modSettings['currentAttachmentUploadDir'])
+				? Config::$modSettings['attachmentUploadDir'][Config::$modSettings['currentAttachmentUploadDir']]
+				: Config::$modSettings['attachmentUploadDir'];
```

Comment:
```suggestion
			$path = (
				!empty(Config::$modSettings['currentAttachmentUploadDir'])
				? Config::$modSettings['attachmentUploadDir'][Config::$modSettings['currentAttachmentUploadDir']]
				: Config::$modSettings['attachmentUploadDir']
			);
```

This is purely cosmetic, but I find it easier to read.